tests/bluetooth/tester/src/audio/btp_bap_broadcast.c

```diff
@@ -1833,6 +1833,66 @@ uint8_t btp_bap_broadcast_assistant_send_past(const void *cmd, uint16_t cmd_len,
 	return BTP_STATUS_SUCCESS;
 }
 
+uint8_t btp_bap_scan_delegator_add_src(const void *cmd, uint16_t cmd_len, void *rsp,
+					    uint16_t *rsp_len)
+{
+	const struct btp_bap_scan_delegator_add_src_cmd *cp = cmd;
+	struct btp_bap_scan_delegator_add_src_rp *rp = rsp;
+	struct bt_bap_scan_delegator_add_src_param param = {0};
+	struct net_buf_simple buf;
+	int err;
+
+	if (cmd_len < sizeof(*cp)) {
+		return BTP_STATUS_FAILED;
+	}
+
+	if (cp->num_subgroups > CONFIG_BT_BAP_BASS_MAX_SUBGROUPS) {
+		return BTP_STATUS_FAILED;
+	}
+
+	bt_addr_le_copy(&param.addr, &cp->broadcaster_address);
+	param.sid = cp->advertiser_sid;
+	param.encrypt_state = (enum bt_bap_big_enc_state)cp->big_encryption;
+	param.broadcast_id = sys_get_le24(cp->broadcast_id);
+	param.num_subgroups = cp->num_subgroups;
+
+	net_buf_simple_init_with_data(&buf, (void *)cp->subgroups, cmd_len - sizeof(*cp));
```

Comment:
The Sonar warning can be ignored for now. I created https://github.com/zephyrproject-rtos/zephyr/issues/91421 a while ago to address this issue with `const` data and `net_buf`